src/main/java/co/inventorsoft/academy/collections/model/Range.java

```diff
@@ -1,60 +1,138 @@
 package co.inventorsoft.academy.collections.model;
 
-import java.util.Collection;
-import java.util.Iterator;
-import java.util.Set;
+import java.util.*;
+import java.util.function.Function;
 
 public class Range<T> implements Set<T> {
 
+    List mylist = new ArrayList();
+
+    public static void main(String[] args) {
+//        Range a = Range.of(1, 10);
+//        a.test();
+//        System.out.println("size "+a.size());
+//        System.out.println("is empty?" + a.isEmpty());
+//        System.out.println(a.add(0.2f));
+//        System.out.println(a.contains(5));
+//        a.test();
+    }
+
+    public Range(float a, float b) {
+        if(!(a==b)){
+            while (a <=b) {
+                mylist.add(a);
+                a = a + 0.1f;
+                a = a * 10;
+                a = Math.round(a);
+                a = a / 10;
+
+            }}
+    }
+
+    public Range(int a, int b) {
+        if (!(a==b)){
+            while (a <= b) {
+                mylist.add(a);
+                a = a + 1;
+            }}
+
+    }
+
+    public static Range of(int i, int i1) {
+        return new Range(i, i1);
+    }
+
+    public static Range of(float i, float i1) {
+        return new Range(i, i1);
+    }
+
+    public static Range of(char i, char i1) {
+        return new Range(i, i1);
+    }
+
+    public static Range<Character> of(char a, char d, Function<Character, Character> characterCharacterFunction) {
```

Comment:
Function is an interface with single method which receives a parameter and returns a value. 

In context of current task received parameter is current value and returned one is next value of the type. 
For integer you pass 3 and should receive 4. 
For character you pass 'a' and should receive 'b'